Source/ChilliSource/Rendering/Base/Renderer.cpp

```diff
@@ -43,55 +66,68 @@ namespace ChilliSource
     
     //------------------------------------------------------------------------------
     Renderer::Renderer() noexcept
-        : m_currentSnapshot(nullptr, Integer2::k_zero, Colour::k_black, RenderCamera())
+        : m_currentMainSnapshot(nullptr, Integer2::k_zero, Colour::k_black, RenderCamera())
+
     {
+
     }
     
     //------------------------------------------------------------------------------
     bool Renderer::IsA(InterfaceIDType interfaceId) const noexcept
     {
         return (Renderer::InterfaceID == interfaceId);
     }
+    
     //------------------------------------------------------------------------------
-    RenderSnapshot Renderer::CreateRenderSnapshot(const Integer2& resolution, const Colour& clearColour, const RenderCamera& renderCamera) noexcept
+    RenderSnapshot Renderer::CreateRenderSnapshot(const RenderTargetGroup* renderTarget, const Integer2& resolution, const Colour& clearColour, const RenderCamera& renderCamera) noexcept
     {
-        auto allocator = m_frameAllocatorQueue.Pop();
-        
-        return RenderSnapshot(allocator, resolution, clearColour, renderCamera);
+        return RenderSnapshot(renderTarget, resolution, clearColour, renderCamera);
     }
     
     //------------------------------------------------------------------------------
-    void Renderer::ProcessRenderSnapshot(RenderSnapshot renderSnapshot) noexcept
+    void Renderer::ProcessRenderSnapshots(IAllocator*&& frameAllocator, RenderSnapshot&& mainRenderSnapshot, std::vector<RenderSnapshot>&& offscreenRenderSnapshots) noexcept
```

Comment:
If you want to make it explicit that the pointer is invalid after the method call just set it to nullptr manually, perhaps with a short note explaining why.